example-app/src/main/java/com/adyen/checkout/example/ui/instant/InstantFragment.kt

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2023 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by ozgur on 3/1/2023.
+ */
+
+package com.adyen.checkout.example.ui.instant
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.core.os.bundleOf
+import androidx.core.view.isVisible
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
+import com.adyen.checkout.components.model.paymentmethods.PaymentMethod
+import com.adyen.checkout.components.model.payments.response.Action
+import com.adyen.checkout.example.databinding.FragmentInstantBinding
+import com.adyen.checkout.example.ui.configuration.CheckoutConfigurationProvider
+import com.adyen.checkout.instant.InstantPaymentComponent
+import com.adyen.checkout.redirect.RedirectComponent
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class InstantFragment : BottomSheetDialogFragment() {
+
+    @Inject
+    internal lateinit var checkoutConfigurationProvider: CheckoutConfigurationProvider
+
+    private var _binding: FragmentInstantBinding? = null
+    private val binding: FragmentInstantBinding
+        get() = requireNotNull(_binding)
+
+    private var instantPaymentComponent: InstantPaymentComponent? = null
+
+    private val instantViewModel: InstantViewModel by viewModels()
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        // Insert return url in extras, so we can access it in the ViewModel through SavedStateHandle
+        arguments = bundleOf().apply {
```

Comment:
Doing it like this is a bit cleaner to me
```Kotlin
val returnUrl = RedirectComponent.getReturnUrl(requireActivity().applicationContext)
arguments = bundleOf(RETURN_URL_EXTRA to returnUrl)
```

Reply:
Indeed. Done